launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -176,6 +178,7 @@ List<String> buildClassPath(String appClassPath) throws IOException {
             "NOTE: SPARK_PREPEND_CLASSES is set, placing locally compiled Spark classes ahead of " +
             "assembly.");
         }
+        boolean isSparkHiveJarAvailable = isSparkHiveJarAvailable(jarsDir);
```

Comment:
It is impossible to detect whether `org.apache.hadoop.hive.ql.plan.FileSinkDesc` can be loaded through methods like `Class.forName`, as it should always return false. This is because the content in the `jarsDir` is not yet in the classpath of the current process.